src/legacy/ui/public/vis/update_status.ts

```diff
@@ -59,8 +59,8 @@ function getUpdateStatus<T extends Status>(
   requiresUpdateStatus: T[] = [],
   obj: any,
   param: { vis: Vis; visData: any; uiState: PersistedState }
-): { [reqStats in T]: boolean } {
-  const status = {} as { [reqStats in T]: boolean };
+): { [reqStats in Status]: boolean } {
```

Comment:
Took a double look at this, I think we'd be better off getting rid of the `T extends Status` bit and just using `Status` in place of `T` everywhere. @timroes do you remember why this uses `T extends Status` instead of just referencing the Status enum directly? The indirection seems to have worked in the past but is now confusing TypeScript which complains about the assignments to `status.aggs` and such below:

![image](https://user-images.githubusercontent.com/1329312/53454430-002c4380-39dc-11e9-868b-aeb6798dcc52.png)

Reply:
Because we know that the return object doesn't have all possible values of the `Status` enum as a key, so `reqStats in Status` is simply wrong. The method signature was in a way so that you pass in an array of enum values as a parameter and the result object will have exactly only those enum values as keys (not more not less from the enum) and boolean value for each of them.

Earlier the correct syntax for that was like that. `[reqStats in Status]` is definitely the wrong typing, since it would say the result object has ALWAYS all the enum values as keys, which it doesn't. I can try to look into that today or tomorrow to find a better solution, that will still break. The least thing that should still work is `{ [reqStats in Status]?: boolean }` which allows each of the keys to be undefined. Since we're losing type information like that, that we're currently having I would like to check if there is another better way to solve that.

Reply:
I tested it. You can simply revert that line again. It's enough if you instantiate the `status` object in the next line as `{ [reqStats in Status]: boolean }` (as you already do). That line can stay the same. And TypeScript won't complain anymore, but will still have the nice syntax for the caller, that the return object will only contain the requested keys.